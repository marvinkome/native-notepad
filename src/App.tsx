/**
 * ./src/App
 *
 * Root component
 */

import { Container } from 'native-base';
import * as React from 'react';
import { StackNavigator } from 'react-navigation';
import EditNote from './components/edit_note';
import Home from './components/home';
import NewNote from './components/new_note';
import Note from './components/note';

export const RootStack = StackNavigator(
    {
        Home: Home,
        Note: Note,
        NewNote: NewNote,
        EditNote: EditNote
    },
    {
        initialRouteName: 'Home',
        navigationOptions: {
            headerTintColor: '#fff',
            headerStyle: {
                backgroundColor: '#607d8b'
            }
        }
    }
);

export default class App extends React.Component {
    render() {
        return <RootStack />;
    }
}
