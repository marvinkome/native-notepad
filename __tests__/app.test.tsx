/**
 * __tests__/app.test
 */

import * as React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';

it('App Renders correctly', () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
});
