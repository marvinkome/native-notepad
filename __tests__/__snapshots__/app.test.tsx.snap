// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "marginTop": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "height": 1334,
              },
              Object {
                "backgroundColor": "#fff",
              },
            ]
          }
          virtual={undefined}
        >
          <RCTScrollView
            automaticallyAdjustContentInsets={false}
            contentContainerStyle={
              Array [
                Object {
                  "padding": undefined,
                },
                undefined,
              ]
            }
            contentInset={
              Object {
                "bottom": 0,
              }
            }
            enableAutomaticScroll={true}
            enableResetScrollToCoords={true}
            extraHeight={75}
            extraScrollHeight={0}
            getScrollResponder={[Function]}
            handleOnScroll={[Function]}
            keyboardDismissMode="interactive"
            keyboardOpeningTime={250}
            keyboardShouldPersistTaps="handled"
            keyboardSpace={0}
            onScroll={[Function]}
            resetKeyboardSpace={[Function]}
            resetScrollToCoords={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            scrollEventThrottle={1}
            scrollForExtraHeightOnAndroid={[Function]}
            scrollToEnd={[Function]}
            scrollToFocusedInput={[Function]}
            scrollToPosition={[Function]}
            showsVerticalScrollIndicator={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
              }
            }
            viewIsInsideTabBar={false}
            virtual={undefined}
          >
            <View>
              <RCTScrollView
                closeOnRowBeginSwipe={false}
                closeOnRowPress={true}
                closeOnScroll={true}
                dataArray={
                  Array [
                    "Hey",
                    "Hello",
                  ]
                }
                dataSource={
                  ListViewDataSource {
                    "items": 2,
                  }
                }
                directionalDistanceChangeThreshold={2}
                disableLeftSwipe={false}
                disableRightSwipe={false}
                enableEmptySections={true}
                leftOpenValue={0}
                previewFirstRow={false}
                recalculateHiddenLayout={false}
                renderRow={[Function]}
                renderScrollComponent={[Function]}
                rightOpenValue={0}
                style={Object {}}
                swipeToOpenPercent={50}
                virtual={undefined}
              >
                <View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={null}
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <View
                      onPress={[Function]}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderBottomColor": "transparent",
                            "borderBottomWidth": 0.5,
                            "flexDirection": "row",
                            "marginLeft": 0,
                            "paddingBottom": 15,
                            "paddingRight": 18,
                            "paddingTop": 15,
                          },
                          Object {
                            "borderLeftColor": "red",
                            "borderLeftWidth": 7,
                            "marginBottom": 1,
                          },
                        ]
                      }
                      virtual={undefined}
                    >
                      <View
                        style={
                          Object {
                            "alignItems": null,
                            "alignSelf": null,
                            "flex": 1,
                          }
                        }
                        virtual={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "Roboto",
                                "fontSize": 16,
                                "marginLeft": 12,
                                "marginRight": 12,
                              },
                              Object {
                                "fontSize": 23,
                                "fontWeight": "600",
                              },
                            ]
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                          Hey
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "flex-end",
                            "alignSelf": "center",
                            "flex": 0.28,
                            "paddingBottom": null,
                            "paddingLeft": null,
                            "paddingRight": null,
                            "paddingTop": null,
                          }
                        }
                        virtual={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "alignSelf": null,
                                "color": "#000",
                                "fontFamily": "Roboto",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 13,
                                "fontWeight": "200",
                              },
                            ]
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                          11/12/18
                        </Text>
                      </View>
                    </View>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={null}
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <View
                      onPress={[Function]}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderBottomColor": "transparent",
                            "borderBottomWidth": 0.5,
                            "flexDirection": "row",
                            "marginLeft": 0,
                            "paddingBottom": 15,
                            "paddingRight": 18,
                            "paddingTop": 15,
                          },
                          Object {
                            "borderLeftColor": "red",
                            "borderLeftWidth": 7,
                            "marginBottom": 1,
                          },
                        ]
                      }
                      virtual={undefined}
                    >
                      <View
                        style={
                          Object {
                            "alignItems": null,
                            "alignSelf": null,
                            "flex": 1,
                          }
                        }
                        virtual={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "Roboto",
                                "fontSize": 16,
                                "marginLeft": 12,
                                "marginRight": 12,
                              },
                              Object {
                                "fontSize": 23,
                                "fontWeight": "600",
                              },
                            ]
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                          Hello
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "flex-end",
                            "alignSelf": "center",
                            "flex": 0.28,
                            "paddingBottom": null,
                            "paddingLeft": null,
                            "paddingRight": null,
                            "paddingTop": null,
                          }
                        }
                        virtual={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "alignSelf": null,
                                "color": "#000",
                                "fontFamily": "Roboto",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 13,
                                "fontWeight": "200",
                              },
                            ]
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                          11/12/18
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </RCTScrollView>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 20,
                  "flexDirection": "column",
                  "height": 56,
                  "left": undefined,
                  "position": "absolute",
                  "right": 20,
                  "top": undefined,
                  "width": 56,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "blue",
                    "borderRadius": 28,
                    "bottom": 0,
                    "elevation": 4,
                    "height": 56,
                    "justifyContent": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.4,
                    "shadowRadius": 2,
                    "width": 56,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#fff",
                          "fontSize": 24,
                          "lineHeight": 27,
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                  virtual={undefined}
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        collapsable={undefined}
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "backgroundColor": "#607d8b",
            "borderBottomColor": "#A7A7AA",
            "borderBottomWidth": 0.5,
            "height": 64,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "flexDirection": "row",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={undefined}
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessibilityTraits="header"
                accessible={true}
                allowFontScaling={true}
                collapsable={undefined}
                ellipsizeMode="tail"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 17,
                    "fontWeight": "700",
                    "marginHorizontal": 16,
                    "textAlign": "center",
                  }
                }
              >
                Home
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
